Fix TestCheckDockerVersion on macos

The unit test `TestCheckDockerVersion` is broken when running on with "Docker for Desktop" docker instance, on MacOS for example.

Expected test output is `official release of Docker` but IRL we get `release of Docker Desktop `

```
--- FAIL: TestCheckDockerVersion (0.00s)
    --- FAIL: TestCheckDockerVersion/checkDockerVersion_test#18 (0.00s)
        docker_test.go:120: Error expected Fix to contain "Install the official release of Docker (Minimum recommended version is 18.09.0, current version is dev)", but got "Install the official release of Docker Desktop (Minimum recommended version is 18.09.0, current version is dev)"
    --- FAIL: TestCheckDockerVersion/checkDockerVersion_test#19 (0.00s)
        docker_test.go:120: Error expected Fix to contain "Install the official release of Docker (Minimum recommended version is 18.09.0, current version is library-import)", but got "Install the official release of Docker Desktop (Minimum recommended version is 18.09.0, current version is library-import)"
    --- FAIL: TestCheckDockerVersion/checkDockerVersion_test#20 (0.00s)
        docker_test.go:120: Error expected Fix to contain "Install the official release of Docker (Minimum recommended version is 18.09.0, current version is foo.bar.baz)", but got "Install the official release of Docker Desktop (Minimum recommended version is 18.09.0, current version is foo.bar.baz)"
FAIL

```

--- a/pkg/minikube/registry/drvs/docker/docker_test.go
+++ b/pkg/minikube/registry/drvs/docker/docker_test.go
@@ -20,6 +20,8 @@ import (
 	"fmt"
 	"strings"
 	"testing"
+
+	"k8s.io/minikube/pkg/minikube/driver"
 )
 
 type testCase struct {
@@ -88,22 +90,22 @@ func TestCheckDockerVersion(t *testing.T) {
 			// "dev" is set when Docker (Moby) was installed with `make binary && make install`
 			version: "linux-dev",
 			expect:  "",
-			expectFixContains: fmt.Sprintf("Install the official release of Docker (Minimum recommended version is %02d.%02d.%d, current version is dev)",
-				minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
+			expectFixContains: fmt.Sprintf("Install the official release of %s (Minimum recommended version is %02d.%02d.%d, current version is dev)",
+				driver.FullName(driver.Docker), minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
 		},
 		{
 			// "library-import" is set when Docker (Moby) was installed with `go build github.com/docker/docker/cmd/dockerd` (unrecommended, but valid)
 			version: "linux-library-import",
 			expect:  "",
-			expectFixContains: fmt.Sprintf("Install the official release of Docker (Minimum recommended version is %02d.%02d.%d, current version is library-import)",
-				minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
+			expectFixContains: fmt.Sprintf("Install the official release of %s (Minimum recommended version is %02d.%02d.%d, current version is library-import)",
+				driver.FullName(driver.Docker), minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
 		},
 		{
 			// "foo.bar.baz" is a triplet that cannot be parsed as "%02d.%02d.%d"
 			version: "linux-foo.bar.baz",
 			expect:  "",
-			expectFixContains: fmt.Sprintf("Install the official release of Docker (Minimum recommended version is %02d.%02d.%d, current version is foo.bar.baz)",
-				minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
+			expectFixContains: fmt.Sprintf("Install the official release of %s (Minimum recommended version is %02d.%02d.%d, current version is foo.bar.baz)",
+				driver.FullName(driver.Docker), minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
 		},
 	}...)
 

